table/format.h

```diff
@@ -76,6 +76,35 @@ class BlockHandle {
   static const BlockHandle kNullBlockHandle;
 };
 
+// Value in block-based table file index.
+//
+// The index entry for block n is: y -> h, [x],
+// where: y is some key between the last key of block n (inclusive) and the
+// first key of block n+1 (exclusive); h is BlockHandle pointing to block n;
+// x, if present, is the first key of block n (unshortened).
+// This struct represents the "h, [x]" part.
+struct IndexValue {
+  BlockHandle handle;
+  // Empty means unknown.
```

Comment:
Keep in mind that, empty string is a valid RocksDB key, so we have a conflict here. I don't think it will cause a bug in your implementation, because it's OK to skip the first key in this case. But it will be nice if we add a test case for it.

Reply:
It's not a valid *internal* key though, right?

Reply:
Right, it's not.